app/models/git_repository.rb

```diff
@@ -0,0 +1,104 @@
+class GitRepository < ApplicationRecord
+  include AuthenticationMixin
+
+  validates :url, :format => URI::regexp(%w(http https)), :allow_nil => false
+
+  default_value_for :verify_ssl, OpenSSL::SSL::VERIFY_PEER
+
+  has_many :git_branches, :dependent => :destroy
+  has_many :git_tags, :dependent => :destroy
+
+  VALID_KEYS = %w(commit_sha commit_message commit_time).freeze
```

Comment:
minor, but VALID_KEYS feels like the wrong name here...I'm thinking INFO_KEYS, since the _info methods use this.